views/js/tools/diagnostic/diagnostic.js

```diff
@@ -728,9 +734,20 @@ define([
                     $btn.addClass('hidden');
                     $result.find('[data-action="show-details"]').removeClass('hidden');
                 });
-            })
-            .init(config);
+            });
+
+        _.defer(function() {
+            diagComponent.init();
+        });
+
+        return diagComponent;
     }
 
-    return diagnosticFactory;
+    /**
+     * Wraps the diagnostic factory to be loaded in diagnostic component as a module
+     */
+    return {
+        name: 'diagnostic',
+        init: diagnosticFactory
```

Comment:
This could be simplified. The name should not be managed here. That introduces complexity and could lead to mapping issues. Moreover, that also introduces unnecessary changes on existing components.